### ISSUE TITLE
Feature: Option for Enum

### ISSUE THREAD

**Issue by Leomaradan** [azat-io/eslint-plugin-perfectionist#3]
### What rule do you want to change?

sort-enums

### Describe the problem

Currently, the sort-enums rule can break the code, if the enums don't use value

In the following code, VAL2 has an implicit value of 0, and VAL1 an implicit value of 1. So, if I sort with the fixer, VAL1 will be set first, and implicit values will be inverted

### Code example

```
export enum Enum1 {
  VAL2, // implicit value: 0
  VAL1, // implicit value: 1
}
```

### Additional comments

IMO, the rule must sort only enums that come with explicit values, and must avoid changing enums with implicit values

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Released in https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v1.0.1

**Comment by u3u** [comment]
https://github.com/azat-io/eslint-plugin-perfectionist/blob/dbfd97b4edc2aeb7bfae475c071cc125caebbb59/rules/sort-enums.ts#L71-L75

I think only when all items are initialized can they be sorted safely.

```ts
/* eslint perfectionist/sort-enums: "warn" */
enum E {
  Z = 1,
  A, // implicit value: 2
}
// ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓ ↓
enum E {
  A, // implicit value: 0
  Z = 1,
}
```

**Comment by azat-io** [comment]
@u3u Fixed in f4a0e25f3c3f763ab1105b7bfa884dde1bd831e2
Thanks!
